src/types/datasource.ts

```diff
@@ -0,0 +1,203 @@
+export const PostgreSQLDBType = t.literal("PostgreSQL");
+export type PostgreSQLDBType = t.TypeOf<typeof PostgreSQLDBType>;
+
+export const CDCConnectionSourceType = t.literal("CDC");
+export type CDCConnectionSourceType = t.TypeOf<typeof CDCConnectionSourceType>;
+export const SelectAllConnectionSourceType = t.literal("SELECT_ALL");
+export type SelectAllConnectionSourceType = t.TypeOf<
+  typeof SelectAllConnectionSourceType
+>;
+
+export const DBDataSourceType = t.literal("DB");
+export type DBDataSourceType = t.TypeOf<typeof DBDataSourceType>;
+
+export const QueueDataSourceType = t.literal("QUEUE");
+export type QueueDataSourceType = t.TypeOf<typeof QueueDataSourceType>;
+
+export const DataSourceConnectionCommon = t.type({
+  connectionString: NonEmptyString,
+});
+export type DataSourceConnectionCommon = t.TypeOf<
+  typeof DataSourceConnectionCommon
+>;
+
+export const CommonDocumentDbCdcProps = t.intersection([
+  t.type({
+    collectionName: NonEmptyString,
+    dbName: NonEmptyString,
+    leaseContainerName: NonEmptyString,
+  }),
+  t.partial({
+    leaseContainerPrefix: NonEmptyString,
+  }),
+]);
+export type CommonDocumentDbCdcProps = t.TypeOf<
+  typeof CommonDocumentDbCdcProps
+>;
+
+export const CosmosCdcProps = t.intersection([
+  CommonDocumentDbCdcProps,
+  t.partial({
+    startingFrom: IsoDateFromString,
+  }),
+]);
+
+export type CosmosCdcProps = t.TypeOf<typeof CosmosCdcProps>;
+
+export const CommonRelationalDbCdcProps = t.type({
+  dbName: NonEmptyString,
+  replicaSlotName: NonEmptyString,
+  subscriptionName: NonEmptyString,
+  tableName: NonEmptyString,
+});
+export type CommonRelationalDbCdcProps = t.TypeOf<
+  typeof CommonRelationalDbCdcProps
+>;
+
+export const CommonCDCDataSource = t.type({
+  sourceType: CDCConnectionSourceType,
+});
+export type CommonCDCDataSource = t.TypeOf<typeof CommonCDCDataSource>;
+
+export const CDCCosmosDataSource = t.exact(
+  t.intersection([
+    CommonCDCDataSource,
+    t.type({
+      dbType: CosmosDBType,
+      props: CosmosCdcProps,
+    }),
+  ]),
+);
+export type CDCCosmosDataSource = t.TypeOf<typeof CDCCosmosDataSource>;
+
+export const CDCMongoDataSource = t.exact(
+  t.intersection([
+    CommonCDCDataSource,
+    t.type({
+      dbType: MongoDBType,
+      props: CommonDocumentDbCdcProps,
+    }),
+  ]),
+);
+export type CDCMongoDataSource = t.TypeOf<typeof CDCMongoDataSource>;
+
+export const CDCPostgresDataSource = t.exact(
+  t.intersection([
+    CommonCDCDataSource,
+    t.type({
+      dbType: PostgreSQLDBType,
+      props: CommonRelationalDbCdcProps,
+    }),
+  ]),
+);
+export type CDCPostgresDataSource = t.TypeOf<typeof CDCPostgresDataSource>;
+
+export const CDCDataSource = t.union([
+  CDCCosmosDataSource,
+  CDCMongoDataSource,
+  CDCPostgresDataSource,
+]);
+export type CDCDataSource = t.TypeOf<typeof CDCDataSource>;
+
+export const CommonDbSelectAllProps = t.type({
+  dbName: NonEmptyString,
+  pageSize: WithinRangeInteger(1, 1000),
```

Comment:
Could this value be optional ?
We can provide a default value if it's not defined.